Mathlib/Algebra/GroupWithZero/Range.lean

```diff
@@ -117,7 +117,47 @@ lemma mem_valueGroup {b : Bˣ} (hb : b.1 ∈ range f) : b ∈ valueGroup f := by
 lemma inv_mem_valueGroup {b : Bˣ} (hb : b.1 ∈ range f) : b⁻¹ ∈ valueGroup f :=
   Subgroup.inv_mem _ (mem_valueGroup f hb)
 
-end MonoidWithZero
+end MonoidWithZeroHom
+
+noncomputable section Restrict
+
+variable [MonoidWithZero A] [GroupWithZero B] [MonoidWithZeroHomClass F A B] {f}
+  [DecidablePred fun b : B ↦ b = 0]
+
+/-- The inclusion of `valueGroup₀ f` into `B` as a multiplicative homomorphism. -/
+@[simps!]
+def valueGroup₀.embedding : ValueGroup₀ f →*₀ B :=
+  MonoidWithZeroHom.comp (WithZero.withZeroUnitsEquiv (G := B))
+    <| WithZero.map' (valueGroup f).subtype
+
+variable (f) in
+/-- This is the restriction of `f` as a function taking values in `valueGroup₀ f`. It cannot land
+in `valueMonoid₀ f` because in general `f a` needs not be a unit, so it will not be in
+`valueMonoid₀ f`. -/
```

Comment:
I don't understand this docstring. It seems to explain why the codomain cannot be `valueMonoid` but not why it cannot be `ValueMonoid₀`

Reply:
You are right, it is an outdated docstring for when I hoped that `B` could be `MonoidWithZero` instead of a `GroupWithZero`. I have shortened it.